src/botPage/bot/Interface/ToolsInterface.js

```diff
@@ -1,13 +1,38 @@
 import CandleInterface from './CandleInterface';
 import MiscInterface from './MiscInterface';
 import IndicatorsInterface from './IndicatorsInterface';
+import { translate } from '../../../common/i18n';
 
 // prettier-ignore
 export default Interface => class extends IndicatorsInterface(
     MiscInterface(CandleInterface(Interface))) {
     getToolsInterface() {
         return {
-            getTime: () => parseInt(new Date().getTime() / 1000),
+            getTime   : () => parseInt(new Date().getTime() / 1000),
+            toDateTime: (timestamp) => {
+                let dateTime;
+                if(typeof timestamp === 'number') {
+                    dateTime = new Date(timestamp * 1000);
+                    return `${[
+                        dateTime.getFullYear().toString(),
+                        dateTime.getMonth()+1 < 10 ? `0${  dateTime.getMonth()+1}` : dateTime.getMonth()+1,
+                        dateTime.getDate() < 10 ? `0${  dateTime.getDate()}` : dateTime.getDate(),
```

Comment:
Because you're making the same calculation in a lot of places you can create a common function that does exactly that. For example:

```JavaScript
const getTwoDigitValue = input => {
    if (input < 10) {
        return `0${input}`;
    }
    return `${input}`;
}
```

For readability you can then put each of the date parts into their own variable:

```JavaScript
const year = dateTime.getFullYear();
const month = getTwoDigitValue(dateTime.getMonth() + 1);
const day = getTwoDigitValue(dateTime.getDate());
const hours = getTwoDigitValue(dateTime.getHours());
const minutes = getTwoDigitValue(dateTime.getMinutes());
const seconds = getTwoDigitValue(dateTime.getSeconds());
```

With these variables you can return a clearer template literal

```JavaScript
return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
```